src/Workspaces/Core/Portable/Utilities/SimpleTaskQueue.cs

```diff
@@ -29,44 +29,60 @@ public SimpleTaskQueue(TaskScheduler taskScheduler)
             _latestTask = Task.CompletedTask;
         }
 
-        private TTask ScheduleTaskWorker<TTask>(Func<int, TTask> taskCreator)
+        private TTask ScheduleTaskWorker<TArg, TTask>(Func<int, TArg, TTask> taskCreator, TArg arg)
             where TTask : Task
         {
             lock (_gate)
             {
                 _taskCount++;
                 var delay = (_taskCount % 100) == 0 ? 1 : 0;
 
-                var task = taskCreator(delay);
+                var task = taskCreator(delay, arg);
 
                 _latestTask = task;
 
                 return task;
             }
         }
 
+        [PerformanceSensitive(
+            "https://developercommunity.visualstudio.com/content/problem/854696/changing-target-framework-takes-10-minutes-with-10.html",
+            AllowCaptures = false)]
         public Task ScheduleTask(Action taskAction, CancellationToken cancellationToken = default)
         {
-            return ScheduleTaskWorker<Task>(delay => _latestTask.ContinueWithAfterDelay(
-                taskAction, cancellationToken, delay, TaskContinuationOptions.None, _taskScheduler));
+            return ScheduleTaskWorker(
+                (delay, arg) => arg.Item1._latestTask.ContinueWithAfterDelay(arg.taskAction, arg.cancellationToken, delay, TaskContinuationOptions.None, arg.Item1._taskScheduler),
+                (this, taskAction, cancellationToken));
```

Comment:
i'm personally not a fan of this approach because (far too often) we've seen the lambdas *still* capturing.  Until static lamdbdas go in, i would actually prefer to see this done with a static peer method, or a static local function.  That way you can be 100% certain the compiler is preventing capturing from acutally happening.

Reply:
This is the pattern we intend for PerformanceSensitiveAnalyzers to recognize and validate.